dotnet/samples/Concepts/ChatCompletion/Connectors_WithMultipleLLMs.cs

```diff
@@ -26,57 +27,92 @@ public async Task RunAsync()
                 serviceId: "OpenAIChat")
             .Build();
 
-        await RunByServiceIdAsync(kernel, "AzureOpenAIChat");
-        await RunByModelIdAsync(kernel, TestConfiguration.OpenAI.ChatModelId);
-        await RunByFirstModelIdAsync(kernel, "gpt-4-1106-preview", TestConfiguration.AzureOpenAI.ChatModelId, TestConfiguration.OpenAI.ChatModelId);
+        // Preconfigured function settings
+        await PreconfiguredFunctionSettingsByFirstModelIdAsync(kernel, ["gpt-4-1106-preview", TestConfiguration.AzureOpenAI.ChatModelId, TestConfiguration.OpenAI.ChatModelId]);
```

Comment:
The names of these methods don't describe what each function does. From reading the code it looks like this method takes a list of modelIds and then executes the prompt with the first matching service. So a name like `InvokePromptWithFirstMatchingModelIdAsync` would be clearer. Each function should also have a comment explaining what it does.

Reply:
Also consider making each of the functions into it's own `Fact` so that they can be run is isolation